packages/flutter/lib/src/foundation/_platform_io.dart

```diff
@@ -38,5 +38,5 @@ platform.TargetPlatform get defaultTargetPlatform {
       'Consider updating the list of TargetPlatforms to include this platform.'
     );
   }
-  return result;
+  return result!;
```

Comment:
there is a null check above this line, it is odd that a `!` is needed. It should be possible for the compiler to understand how the throw affects the control flow here.

Reply:
I had the same feeling when I reported dart-lang/sdk#42380 but it is needed.